doc/specs/#4191 - Formatted Copy/spec.md

```diff
@@ -0,0 +1,90 @@
+---
+author: Kayla Cinnamon @cinnamon-msft
+created on: 2020-04-01
+last updated: 2020-04-07
+issue id: #4191
+---
+
+# Formatted Copy
+
+## Abstract
+
+When copying text, the Terminal should provide the option of including formatting. Not all apps that receive text allow for picking which format you want when pasting. The default should be to only copy plain text, based on the response from this poll on Twitter.
+
+![Twitter poll](twitter-poll.png)
+
+## Solution Proposals
+
+A proposal for the right click behavior as well as two user settings proposals are described below:
+
+1. [Right click behavior](#right-click-behavior)
+2. [Settings option 1 - global setting](#settings-option-1---global-setting)
+3. [Settings option 2 - key binding argument](#settings-option-2---key-binding-argument)
+
+### Right click behavior
+
+By default, right clicking to copy would only copy the plain text. If the user would like to copy the formatting, they can hold `alt` and right click (this behavior is the same in conhost).
+
+### Settings option 1 - global setting
+
+We could have a global setting that adds the formatting when the user copies. When set, this would change the right click behavior.
+
+### Settings option 2 - key binding argument
+
+We could add an argument to the `copy` key binding argument to allow for formatted copying when the user chooses to do so.
+
+## UI/UX Design
+
+### Settings option 1 - global setting
+
+a. The user could list which kinds of formats they want included when they copy. When right clicking, they would copy with these formats. If this is set to something other than `plain`, holding `alt` and right clicking would copy plain text.
+
+`"copyFormats": ["html","rtf","plain"]`
+
+b. We could also just combine html and rtf into a single boolean. Users would either get plain text only (`false`) or all formatting (`true`) onto their clipboard. If this is set to `true`, the default right click behavior is reversed: right clicking copies the formatting and holding `alt` copies only the plain text.

```

Comment:
Correct me if I'm wrong, but if option b is the one we're going for, we're _not_ having the `Alt-RightClick` to copy plaintext/formatting right? Right click should just copy formatting if the setting is true, and plain text if false.

Reply:
Ah nice catch. I just removed references to `alt` right click functionality. 👍